### ISSUE TITLE
activate/conda/deactivate scripts created when they shouldn't be...

### ISSUE THREAD

**Issue by mcg1969** [conda/conda#3808]
I have a package I'm building called Burger (*) sitting in my home directory: `/Users/mcg/burger/`, and given standard Python packaging structure there's also a directory `/Users/mcg/burger/burger/`.

Being the smart person I am I create a separate conda environment for my project called, you guessed it, `burger`. So when I'm ready to develop I do
```
cd ~/burger
source activate burger
```
Every once in awhile my twitchy fingers try to autocomplete that last `burger` and I get this:
```
cd ~/burger
source activate burger/
```
This, of course, doesn't do what I want: it activates a conda environment _that doesn't exist_, in the location `/Users/mcg/burger/burger/`. When things don't work right I realize my mistake and do `source activate burger` and all is well.

Or is it? Alas, that incorrect `activate` command left me a little present: a directory `/Users/mcg/burger/burger/bin` containing symbolic links to `activate`, `deactivate`, and `conda`. 

They just sit there, lurking quietly, but not really doing anything bad... unless I accidentally check them into my repo, in which case they [break my CI builds](https://github.com/conda/conda-build/issues/1509).

I would suggest that the script that creates these symbolic links should do at least a _cursory_ check to confirm that the environment it is activating is legitimate. If it is not, it should  issue an error. The consequences of not doing so seem subtle and usually benign, but take my word for it, they're at times quite vexing.

_(*) the name of the actual package has been changed to protect the innocent._

**Comment by mcg1969** [comment]
(P.S. I'm hungry for some reason)

**Comment by njalerikson** [comment]
OK so there is no real way to check whether the "environment" that a user attempts to activate is actually a conda environment or just a mischievous directory.

Of course we can start talking about bad development practice to call so many "different" things `burger`. But clearly that is besides the point here.

The reason there is no current sanity checks is because conda environments do not have to live in any specific location (a user is perfectly allowed to make a environment in any location not in `~/anaconda/envs/`). Plus there is nothing explicitly "conda" about a conda environment (e.g. there is no directory or file similar to `.git` in git repos).

This may suggest that it would be smart to introduce a change to include a `.conda` directory in every conda environment such that a sanity check can be performed.

**Comment by kalefranz** [comment]
Yeah the problem here is confusion of *named environments* versus *environments that have locations*.  I'm not a fan of how this particularly piece was originally designed.  In your situation, it's right now impossible to disambiguate between the two.

> This may suggest that it would be smart to introduce a change to include a .conda directory in every conda environment such that a sanity check can be performed.

I have plans to do something like this.  With #3175 merged we can start to sketch out where we go next.

**Comment by kalefranz** [comment]
> I would suggest that the script that creates these symbolic links should do at least a cursory check to confirm that the environment it is activating is legitimate.

What *is* a legitimate environment?  Ilan is pretty adamant that an empty directory is a legitimate environment.  Which leaves us about at ¯\\\_(ツ)\_/¯

**Comment by kalefranz** [comment]
Fixed with #5044 in conda 4.4.
